wpiutil/src/main/native/cpp/DataLog.cpp

```diff
@@ -618,6 +619,106 @@ void DataLog::AppendStringArray(int entry,
   }
 }
 
+template <typename V1, typename V2>
+inline bool UpdateImpl(std::optional<std::vector<V1>>& lastValue,
+                       std::span<const V2> data) {
+  if (!lastValue || !std::equal(data.begin(), data.end(), lastValue->begin(),
+                                lastValue->end())) {
+    if (lastValue) {
+      lastValue->assign(data.begin(), data.end());
+    } else {
+      lastValue = std::vector<V1>{data.begin(), data.end()};
+    }
+    return true;
+  }
+  return false;
+}
+
+template <typename V1>
+inline bool UpdateImpl(std::optional<std::vector<V1>>& lastValue,
+                       std::span<const bool> data) {
+  if (!lastValue ||
+      !std::equal(data.begin(), data.end(), lastValue->begin(),
+                  lastValue->end(), [](auto a, auto b) { return a == b; })) {
```

Comment:
```suggestion
                  lastValue->end(), [](auto a, bool b) { return static_cast<bool>(a) == b; })) {
```
MSVC warns about comparing an integer to boolean ([C4085](https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4805?view=msvc-170)) because the result can be surprising when the integer is not one of 0 or 1. (e.g., 2 != true since true is treated as 1.) It seems the best fix is to explicitly cast the int to bool before doing the conversion. (We could also cast the bool to int, which would have different behavior if we have values in `lastValue` besides 0 and 1, which could happen via the `Update(std::span<const int>, int64_t)` overload.)
The exact way you do this is up to you (e.g., `!!a` instead of a static cast, maybe leaving `b` as `auto` instead of `bool`), but this is one way to fix the warning and be somewhat self-explanatory.